### PR TITLE
User authentication and authorization

### PR DESCRIPTION
**Pull request by shashankshk**
1. Added bcrypt and JWT.
2. Added user login and signup.

### CODE REVIEW

**Review by Ronnie-wof — APPROVED** [review]

File: requirements.txt [hunk 1]
```diff
@@ -2,4 +2,6 @@ Django
 djangorestframework
 django-cors-headers
 drf-yasg
-pylint
\ No newline at end of file
+pylint
+psycopg2
+bcrypt==3.2.0
```
Comment (n-kumawat) [inline requirements.txt]: Add 'djangorestframework-simplejwt' as well. The build shows import error.

File: asset_mappr/settings.py [hunk 2]
```diff
@@ -20,70 +21,78 @@
...
-    'default': {
-        'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': BASE_DIR / 'db.sqlite3',
+    "default": {
+        "ENGINE": "django.db.backends.postgresql",
+        "NAME": "AssetMappr",
+        "USER": "postgres",
+        "PASSWORD": "",
+        "HOST": "127.0.0.1",
+        "PORT": "5432",
```
Comment (n-kumawat) [inline asset_mappr/settings.py]: Consider providing these as environment variables in a .env file with default values

File: asset_mappr/settings.py [hunk 3]
```diff
@@ -20,70 +21,78 @@
 # See https://docs.djangoproject.com/en/4.2/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = 'django-insecure-7ohn3+896-evxn%dr+!4e@fnf+zqg%9nnd$mdg4ojdg9zafrsq'
+SECRET_KEY = "django-insecure-7ohn3+896-evxn%dr+!4e@fnf+zqg%9nnd$mdg4ojdg9zafrsq"
```
Comment (n-kumawat) [inline asset_mappr/settings.py]: Secret must not be included here. Need to change and take as an environment variable

File: asset_mappr/settings.py [hunk 4]
```diff
@@ -93,26 +102,48 @@
...
     },
 ]
 
+# Password hashers
+PASSWORD_HASHERS = [
+    "django.contrib.auth.hashers.BCryptSHA256PasswordHasher",
+    "django.contrib.auth.hashers.PBKDF2PasswordHasher",
+    "django.contrib.auth.hashers.PBKDF2SHA1PasswordHasher",
+    "django.contrib.auth.hashers.Argon2PasswordHasher",
+    "django.contrib.auth.hashers.ScryptPasswordHasher",
```
Comment (n-kumawat) [inline asset_mappr/settings.py:118]: Is the application using all of these hashers?

File: asset_mappr/settings.py [hunk 5]
```diff
@@ -93,26 +102,48 @@
...
+    "DEFAULT_AUTHENTICATION_CLASSES": [
+        "rest_framework_simplejwt.authentication.JWTAuthentication",
+    ],
+}
+
+SIMPLE_JWT = {
+    "ACCESS_TOKEN_LIFETIME": timedelta(
+        minutes=15
+    ),  # Adjust the token lifetime as per your requirements
+    "REFRESH_TOKEN_LIFETIME": timedelta(days=1),
```
Comment (n-kumawat) [inline asset_mappr/settings.py]: Can consider making the timedelta configurable

**Review by n-kumawat — APPROVED** [review]

**Pull request merged**